### PR TITLE
JobChange: Added command to change both jobs at once and prefer dw subjobs on reset

### PR DESCRIPTION
**Pull request by AkadenTK**

### CODE REVIEW

File: addons/JobChange/jobchange.lua [hunk 1]
```diff
@@ -27,119 +27,181 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local main = nil
+    local sub = nil
+    if command:lower() == 'main' then
+        main = job
+        if main and main:upper() == p.main_job then main = nil end
```
Comment (z16) [inline addons/JobChange/jobchange.lua:114]: `main` here must always exist (so does `sub` below), can be removed from the check. Also, inline `if`, ew :<

File: addons/JobChange/jobchange.lua [hunk 2]
```diff
@@ -27,119 +27,181 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+windower.register_event('addon command', function(command, ...)
+    local p = windower.ffxi.get_player()
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local main = nil
+    local sub = nil
+    if command:lower() == 'main' then
```
Comment (z16) [inline addons/JobChange/jobchange.lua:112]: `command` should be lowercased at the start, so it does not need to be done in every check. I would also uppercase the job strings, so it does not need to be performed everywhere below. Would make the code more readable imo, but up to you.

File: addons/JobChange/jobchange.lua [hunk 3]
```diff
@@ -27,119 +27,181 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+        main = job
+        if main and main:upper() == p.main_job then main = nil end
+    elseif command:lower() == 'sub' then
+        sub = job
+        if sub and sub:upper() == p.sub_job then main = nil end
+    elseif command:lower() == 'reset' then
+        log('Resetting Job')
+        sub = p.sub_job:lower()
+    elseif command:contains('/') or command:contains('\\') then
```
Comment (z16) [inline addons/JobChange/jobchange.lua:121]: You allow backslashes for job delimiters? Why?

File: addons/JobChange/jobchange.lua [hunk 4]
```diff
@@ -27,119 +27,181 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+                log('Conflict with '..(change.is_main and 'main' or 'sub')..' job. Changing to: '..res.jobs[change.job_id].ens)
+            else
+                log('Changing '..(change.is_main and 'main' or 'sub')..' job to: '..res.jobs[change.job_id].ens)
+            end
+            jobchange(change.job_id, change.is_main)
+
+            coroutine.sleep(0.5)
+        end
+    else
+        log('Not close enough to a Moogle!')
```
Comment (z16) [inline addons/JobChange/jobchange.lua:205]: This case could be checked for at the very start to save some otherwise unnecessary calculations.

**Review by z16 — APPROVED** [review]
Just left a few comments, looks fine though. Let me know if you want to adjust some of them or to pull it as is.

File: addons/JobChange/jobchange.lua (hunk 3 above)
Comment (AkadenTK) [reply addons/JobChange/jobchange.lua:121]: Why not?

**Comment by AkadenTK** [comment]
I agree with all the comments, but don't have an active sub to test properly (even though they're very basic changes). I've been using it pretty consistently though until I unsubbed. I'd be okay with it as is.

File: addons/JobChange/jobchange.lua (hunk 3 above)
Comment (z16) [reply addons/JobChange/jobchange.lua:121]: Added complexity, lower performance, more room for bugs, and no one uses it :)

**Pull request merged**